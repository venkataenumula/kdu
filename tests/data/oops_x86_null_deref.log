BUG: unable to handle kernel NULL pointer dereference at 0000000000000000
IP: [<ffffffffa03e1012>] my_oops_init+0x12/0x40 [oops]
PGD 0 P4D 0 
Oops: 0002 [#1] SMP PTI
CPU: 1 PID: 2248 Comm: insmod Tainted: P           O      5.15.0-kdu #1
Hardware name: KDU Labs KDU-Box/Rev1, BIOS 1.0.0 01/01/2024
RIP: 0010:[<ffffffffa03e1012>]  [<ffffffffa03e1012>] my_oops_init+0x12/0x40 [oops]
RSP: 0018:ffffb123c0f37d98  EFLAGS: 00010282
RAX: 0000000000000000 RBX: ffffffffa03e1000 RCX: 0000000000000001
RDX: 0000000000000000 RSI: 0000000000000000 RDI: 0000000000000000
RBP: ffffb123c0f37da8 R08: 0000000000000000 R09: 0000000000000000
R10: 0000000000000000 R11: 0000000000000000 R12: 0000000000000000
R13: 0000000000000000 R14: 0000000000000000 R15: 0000000000000000
FS:  00007f2b2b9f9740(0000) GS:ffff97eb1ed00000(0000) knlGS:0000000000000000
CS:  0010 DS: 0000 ES: 0000 CR0: 0000000080050033
CR2: 0000000000000000 CR3: 0000000000001000 CR4: 00000000003706e0
Call Trace:
 [<ffffffff810a3d49>] do_one_initcall+0x59/0x1b0
 [<ffffffff8111a78d>] do_init_module+0x5d/0x1f0
 [<ffffffff8111aa01>] load_module+0x201/0x2100
 [<ffffffff8108f3c1>] __x64_sys_init_module+0x151/0x190
 [<ffffffff8100236b>] do_syscall_64+0x3b/0x90
 [<ffffffff81c0009c>] entry_SYSCALL_64_after_hwframe+0x44/0xae
Code: 90 90 90 90 c7 04 25 00 00 00 00 00 00 00 00  <c7> 04 25 00 00 00 00 00 00 00 00  c3 90 90 90
