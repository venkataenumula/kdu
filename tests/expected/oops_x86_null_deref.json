{
  "arch": "x86_64",
  "type": "oops",
  "fault": {
    "desc": "page fault",
    "error_code_hex": "0002",
    "decode": {
      "present": false,
      "write": true,
      "user": false
    },
    "cr2": "0x0"
  },
  "taint": "P",
  "pc": {
    "addr": "0xffffffffa03e1012",
    "symbol": "my_oops_init",
    "offset_hex": "0x12",
    "system_map_match": true
  },
  "call_trace": [
    {
      "addr": "0xffffffff810a3d49",
      "symbol": "do_one_initcall",
      "approx": true
    },
    {
      "addr": "0xffffffff8111a78d",
      "symbol": "do_init_module",
      "approx": true
    },
    {
      "addr": "0xffffffff8111aa01",
      "symbol": "load_module",
      "approx": true
    },
    {
      "addr": "0xffffffff8108f3c1",
      "symbol": "__x64_sys_init_module",
      "approx": true
    },
    {
      "addr": "0xffffffff8100236b",
      "symbol": "do_syscall_64",
      "approx": true
    },
    {
      "addr": "0xffffffff81c0009c",
      "symbol": "entry_SYSCALL_64_after_hwframe",
      "approx": true
    }
  ],
  "conclusion": "Likely NULL pointer dereference during module init"
}